src/Nethermind/Nethermind.Vault/VaultWallet.cs

```diff
@@ -0,0 +1,121 @@
+        }
+        
+        public async Task<Address[]> GetAccounts()
+        {
+            accounts = new ConcurrentDictionary<Address, string>();
+            var args = new Dictionary<string, object> {};
+
+            string vault = await _vaultManager.SetWalletVault(_vaultConfig.VaultId);
+            var result = await _initVault.ListVaultKeys(_vaultConfig.Token, vault, args);
+            dynamic keys  = JsonConvert.DeserializeObject(result.Item2);
+            foreach(var key in keys)
+            {
+                try 
+                {
+                    string address = Convert.ToString(key.address);
+                    // adds to accounts dict to have addresses assigned to key id's
+                    string keyId = Convert.ToString(key.id);
+                    accounts.TryAdd(new Address(address), keyId);
+                    accounts.Keys.ToArray();
+                } 
+                catch (ArgumentNullException) {}
+            }
+            return accounts.Keys.ToArray();
+        }
+
+        public async Task<Address> NewAccount(KeyArgs args)
+        {
+            // creates default KeyArgs in case of null input
+            if (args == null) 
+            {
+                args = new KeyArgs();
+                args.Name = "name";
+                args.Description = "description";
+                args.Type = "asymmetric";
+                args.Spec = "secp256k1";
+                args.Usage = "sign/verify";
+            }
+            string vault = await _vaultManager.SetWalletVault(_vaultConfig.VaultId);
+            var result = await _initVault.CreateVaultKey(_vaultConfig.Token, vault, args.ToDictionary());
+            dynamic key  = JsonConvert.DeserializeObject(result.Item2);
+            string address = Convert.ToString(key.address);
+            var account = new Address(address);
+            accounts.TryAdd(account, Convert.ToString(key.id));
+            return account;
+        }
+        public async Task DeleteAccount(Address address)
+        {
+            string vault = await _vaultManager.SetWalletVault(_vaultConfig.VaultId);
+            string keyId = await GetKeyIdByAddress(address);
+            var result = await _initVault.DeleteVaultKey(_vaultConfig.Token, vault, keyId);
+        }
+
+        public async Task<Signature> Sign(Address address, Keccak message)
+        {
+            string vault = await _vaultManager.SetWalletVault(_vaultConfig.VaultId);
+            string keyId = await GetKeyIdByAddress(address);
+            var result = await _initVault.SignMessage(_vaultConfig.Token, vault, keyId, message.ToString());   
+            dynamic sig  = JsonConvert.DeserializeObject(result.Item2);
+            string signature = Convert.ToString(sig.signature);
+            
+            return new Signature(signature);
+        }
+
+        public Task<Address> NewAccount(Dictionary<string, object> parameters)
+        {
```

Comment:
KeyArgs keyArgs = parameters["keyArgs"] as KeyArgs;

Reply:
if(!parameters.ContainsKey("keyArgs") throw new InvalidArgumentException(nameof(parameters))

Reply:
var result = await _initVault.CreateVaultKey(_vaultConfig.Token, vaultId, args.ToDictionary());   

and so on

Reply:
Sorry for this ugliness with the dicts. Promise to refactor with proper types soon :)